presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -309,35 +322,39 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                                 "Filtered aggregations not supported with DISTINCT: '%s'",
                                 node);
                     }
+
+                    // ensure that no output fields are referenced from ORDER BY clause
+                    node.getArguments().stream().forEach(expression -> process(expression, true));
```

Comment:
The fact you purposefully ignore outcome of recursive call to `process` is quite non-obvious (`forEach` vs `allMatch`). Maybe + a comment?

Second, AggregationAnalyzer's disagreement is only sometimes signaled by `false` result. Sometimes it's raised as `SemanticException`. You ignore the former but not the latter. Is that purposeful too?